In [105]:
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import pandas as pd
from statsmodels.stats.weightstats import ttest_ind
import pingouin as pg

In [106]:
data_df = pd.read_csv('movieReplicationSet.csv')
data_df

The Life of David Gale (2003)  Wing Commander (1999)  \
0                               NaN                    NaN   
1                               NaN                    NaN   
2                               NaN                    NaN   
3                               NaN                    NaN   
4                               NaN                    NaN   
...                             ...                    ...   
1092                            NaN                    NaN   
1093                            3.0                    4.0   
1094                            NaN                    NaN   
1095                            NaN                    NaN   
1096                            NaN                    NaN   

      Django Unchained (2012)  Alien (1979)  \
0                         4.0           NaN   
1                         1.5           NaN   
2                         NaN           NaN   
3                         2.0           NaN   
4                         3.5           NaN   
...                       ...           ...   
1092                      NaN           NaN   
1093                      NaN           NaN   
1094                      NaN           NaN   
1095                      NaN           NaN   
1096                      4.0           NaN   

      Indiana Jones and the Last Crusade (1989)  Snatch (2000)  \
0                                           3.0            NaN   
1                                           NaN            NaN   
2                                           NaN            NaN   
3                                           3.0            NaN   
4                                           0.5            NaN   
...                                         ...            ...   
1092                                        3.5            NaN   
1093                                        4.0            4.0   
1094                                        NaN            NaN   
1095                                        NaN            NaN   
1096                                        2.5            NaN   

      Rambo: First Blood Part II (1985)  Fargo (1996)  \
0                                   NaN           NaN   
1                                   NaN           NaN   
2                                   NaN           NaN   
3                                   NaN           NaN   
4                                   0.5           1.0   
...                                 ...           ...   
1092                                NaN           NaN   
1093                                2.5           NaN   
1094                                NaN           3.5   
1095                                NaN           NaN   
1096                                NaN           3.0   

      Let the Right One In (2008)  Black Swan (2010)  ...  \
0                             NaN                NaN  ...   
1                             NaN                NaN  ...   
2                             NaN                NaN  ...   
3                             NaN                4.0  ...   
4                             NaN                0.0  ...   
...                           ...                ...  ...   
1092                          NaN                NaN  ...   
1093                          3.5                3.5  ...   
1094                          NaN                NaN  ...   
1095                          NaN                NaN  ...   
1096                          NaN                3.5  ...   

      When watching a movie I cheer or shout or talk or curse at the screen  \
0                                                   1.0                       
1                                                   3.0                       
2                                                   5.0                       
3                                                   3.0                       
4                                                   2.0                       
...        

# q5

In [107]:
only_child = data_df.loc[data_df['Are you an only child? (1: Yes; 0: No; -1: Did not respond)'] == 1, 'The Lion King (1994)'].values

In [108]:
ii = np.isfinite(only_child)
only_child = only_child[ii]
only_child

array([4. , 4. , 4. , 3.5, 3.5, 4. , 4. , 4. , 3.5, 4. , 2. , 4. , 4. ,
       4. , 4. , 4. , 3.5, 3. , 2.5, 3. , 4. , 3. , 4. , 4. , 1. , 2. ,
       4. , 3. , 3.5, 1.5, 3.5, 3.5, 4. , 3. , 3. , 3. , 0.5, 4. , 4. ,
       4. , 3.5, 4. , 4. , 3. , 3. , 3. , 3. , 4. , 3.5, 3.5, 4. , 2.5,
       4. , 3. , 4. , 4. , 4. , 3. , 3.5, 3. , 1.5, 3. , 2.5, 2.5, 4. ,
       4. , 4. , 3.5, 2.5, 4. , 4. , 3.5, 2. , 0.5, 3.5, 3. , 4. , 4. ,
       3.5, 4. , 4. , 2.5, 2. , 3.5, 4. , 2.5, 4. , 4. , 4. , 3. , 4. ,
       0. , 2.5, 3. , 4. , 3.5, 4. , 4. , 3.5, 4. , 3. , 4. , 3. , 2. ,
       4. , 3. , 3. , 4. , 3.5, 4. , 3.5, 3. , 3. , 3.5, 3. , 0.5, 4. ,
       4. , 4. , 2.5, 4. , 3. , 4. , 3.5, 3.5, 3.5, 4. , 2. , 4. , 3.5,
       4. , 3. , 3. , 3. , 4. , 2.5, 4. , 4. , 4. , 4. , 2.5, 3.5, 4. ,
       4. , 3. , 2.5, 3. , 3.5, 4. , 3.5, 3. ])

In [109]:
np.mean(only_child)

3.347682119205298

In [110]:
siblings = data_df.loc[data_df['Are you an only child? (1: Yes; 0: No; -1: Did not respond)'] == 0, 'The Lion King (1994)'].values

In [111]:
ii = np.isfinite(siblings)
siblings = siblings[ii]
siblings

array([4. , 3. , 3.5, 4. , 4. , 4. , 4. , 3.5, 4. , 4. , 4. , 4. , 4. ,
       4. , 4. , 4. , 3.5, 1.5, 2.5, 4. , 3. , 3. , 4. , 3.5, 4. , 4. ,
       4. , 4. , 4. , 3.5, 2.5, 3. , 3. , 4. , 4. , 4. , 3.5, 3.5, 3.5,
       3.5, 3.5, 0.5, 3. , 2.5, 2.5, 4. , 3.5, 0. , 4. , 3. , 3.5, 4. ,
       4. , 4. , 3.5, 4. , 3. , 4. , 4. , 2.5, 4. , 3.5, 3. , 3.5, 2.5,
       4. , 4. , 3.5, 3.5, 4. , 3. , 4. , 4. , 3.5, 3.5, 1.5, 2. , 2. ,
       3.5, 3. , 3. , 4. , 2.5, 3.5, 4. , 4. , 1.5, 4. , 4. , 3.5, 3. ,
       4. , 4. , 3.5, 4. , 4. , 4. , 4. , 4. , 4. , 4. , 4. , 3.5, 4. ,
       3. , 2.5, 4. , 3. , 3. , 4. , 3. , 2.5, 0.5, 4. , 4. , 3.5, 3. ,
       4. , 2.5, 4. , 4. , 1.5, 4. , 4. , 3.5, 4. , 3.5, 3.5, 4. , 3.5,
       4. , 3. , 4. , 4. , 4. , 3. , 3. , 4. , 4. , 4. , 2. , 2.5, 3. ,
       4. , 2.5, 4. , 4. , 2.5, 4. , 3. , 3.5, 3.5, 4. , 3.5, 3. , 3.5,
       4. , 4. , 3. , 3.5, 3.5, 4. , 3. , 3.5, 4. , 3.5, 4. , 4. , 4. ,
       4. , 3.5, 2. , 3. , 4. , 4. , 4. , 4. , 3.5, 4. , 4. , 3.

In [112]:
np.mean(siblings)

3.481958762886598

In [113]:
ttest_ind(only_child, siblings)

(-2.0538889960589852, 0.040267055262682705, 925.0)

In [114]:
pg.ttest(only_child, siblings, correction=False) #Welch’s t-test as the sample size is not equal

T  dof alternative     p-val           CI95%   cohen-d   BF10  \
T-test -2.053889  925   two-sided  0.040267  [-0.26, -0.01]  0.182683  0.765   

           power  
T-test  0.536599

# Two-Sample T-Test with Statsmodels
The p-value of the test comes out to be equal to 0.040, which is greater than the significance level alpha (that is, 0.005). This implies that we can say that the people who are only children enjoy ‘The Lion King (1994)’ is statistically not different from the the people who have siblings. 



In [115]:
only_child

array([4. , 4. , 4. , 3.5, 3.5, 4. , 4. , 4. , 3.5, 4. , 2. , 4. , 4. ,
       4. , 4. , 4. , 3.5, 3. , 2.5, 3. , 4. , 3. , 4. , 4. , 1. , 2. ,
       4. , 3. , 3.5, 1.5, 3.5, 3.5, 4. , 3. , 3. , 3. , 0.5, 4. , 4. ,
       4. , 3.5, 4. , 4. , 3. , 3. , 3. , 3. , 4. , 3.5, 3.5, 4. , 2.5,
       4. , 3. , 4. , 4. , 4. , 3. , 3.5, 3. , 1.5, 3. , 2.5, 2.5, 4. ,
       4. , 4. , 3.5, 2.5, 4. , 4. , 3.5, 2. , 0.5, 3.5, 3. , 4. , 4. ,
       3.5, 4. , 4. , 2.5, 2. , 3.5, 4. , 2.5, 4. , 4. , 4. , 3. , 4. ,
       0. , 2.5, 3. , 4. , 3.5, 4. , 4. , 3.5, 4. , 3. , 4. , 3. , 2. ,
       4. , 3. , 3. , 4. , 3.5, 4. , 3.5, 3. , 3. , 3.5, 3. , 0.5, 4. ,
       4. , 4. , 2.5, 4. , 3. , 4. , 3.5, 3.5, 3.5, 4. , 2. , 4. , 3.5,
       4. , 3. , 3. , 3. , 4. , 2.5, 4. , 4. , 4. , 4. , 2.5, 3.5, 4. ,
       4. , 3. , 2.5, 3. , 3.5, 4. , 3.5, 3. ])

# q6

In [52]:
only_child_ratings = data_df[data_df['Are you an only child? (1: Yes; 0: No; -1: Did not respond)'] == 1].iloc[:,0:400]
only_child_ratings

The Life of David Gale (2003)  Wing Commander (1999)  \
2                               NaN                    NaN   
4                               NaN                    NaN   
5                               3.0                    NaN   
22                              2.5                    3.0   
33                              NaN                    NaN   
...                             ...                    ...   
1058                            3.5                    3.0   
1068                            NaN                    NaN   
1075                            NaN                    NaN   
1076                            NaN                    NaN   
1088                            NaN                    NaN   

      Django Unchained (2012)  Alien (1979)  \
2                         NaN           NaN   
4                         3.5           NaN   
5                         NaN           2.5   
22                        2.5           3.0   
33                        NaN           NaN   
...                       ...           ...   
1058                      4.0           3.5   
1068                      NaN           NaN   
1075                      4.0           4.0   
1076                      NaN           NaN   
1088                      3.0           NaN   

      Indiana Jones and the Last Crusade (1989)  Snatch (2000)  \
2                                           NaN            NaN   
4                                           0.5            NaN   
5                                           2.0            NaN   
22                                          2.5            1.5   
33                                          NaN            NaN   
...                                         ...            ...   
1058                                        4.0            NaN   
1068                                        NaN            NaN   
1075                                        NaN            2.5   
1076                                        4.0            NaN   
1088                                        1.5            NaN   

      Rambo: First Blood Part II (1985)  Fargo (1996)  \
2                                   NaN           NaN   
4                                   0.5           1.0   
5                                   NaN           NaN   
22                                  3.0           3.0   
33                                  NaN           NaN   
...                                 ...           ...   
1058                                NaN           NaN   
1068                                NaN           NaN   
1075                                2.0           3.5   
1076                                NaN           NaN   
1088                                NaN           NaN   

      Let the Right One In (2008)  Black Swan (2010)  ...  X-Men 2 (2003)  \
2                             NaN                NaN  ...             NaN   
4                             NaN                0.0  ...             2.5   
5                             NaN                4.0  ...             4.0   
22                            2.0                2.5  ...             4.0   
33                            NaN                3.5  ...             4.0   
...                           ...                ...  ...             ...   
1058                          3.0                4.0  ...             NaN   
1068                          NaN                NaN  ...             NaN   
1075                          NaN                4.0  ...             4.0   
1076                          NaN                3.0  ...             NaN   
1088                          NaN                2.5  ...             0.5   

      The Usual Suspects (1995)  The Mask (1994)  Jaws (1975)  \
2                           NaN              NaN          NaN   
4                           NaN              3.0          NaN   
5                           NaN              NaN          NaN   
22                          3.5  

In [53]:
only_child_ratings.reset_index(inplace = True)

In [54]:
only_child_ratings

index  The Life of David Gale (2003)  Wing Commander (1999)  \
0        2                            NaN                    NaN   
1        4                            NaN                    NaN   
2        5                            3.0                    NaN   
3       22                            2.5                    3.0   
4       33                            NaN                    NaN   
..     ...                            ...                    ...   
172   1058                            3.5                    3.0   
173   1068                            NaN                    NaN   
174   1075                            NaN                    NaN   
175   1076                            NaN                    NaN   
176   1088                            NaN                    NaN   

     Django Unchained (2012)  Alien (1979)  \
0                        NaN           NaN   
1                        3.5           NaN   
2                        NaN           2.5   
3                        2.5           3.0   
4                        NaN           NaN   
..                       ...           ...   
172                      4.0           3.5   
173                      NaN           NaN   
174                      4.0           4.0   
175                      NaN           NaN   
176                      3.0           NaN   

     Indiana Jones and the Last Crusade (1989)  Snatch (2000)  \
0                                          NaN            NaN   
1                                          0.5            NaN   
2                                          2.0            NaN   
3                                          2.5            1.5   
4                                          NaN            NaN   
..                                         ...            ...   
172                                        4.0            NaN   
173                                        NaN            NaN   
174                                        NaN            2.5   
175                                        4.0            NaN   
176                                        1.5            NaN   

     Rambo: First Blood Part II (1985)  Fargo (1996)  \
0                                  NaN           NaN   
1                                  0.5           1.0   
2                                  NaN           NaN   
3                                  3.0           3.0   
4                                  NaN           NaN   
..                                 ...           ...   
172                                NaN           NaN   
173                                NaN           NaN   
174                                2.0           3.5   
175                                NaN           NaN   
176                                NaN           NaN   

     Let the Right One In (2008)  ...  X-Men 2 (2003)  \
0                            NaN  ...             NaN   
1                            NaN  ...             2.5   
2                            NaN  ...             4.0   
3                            2.0  ...             4.0   
4                            NaN  ...             4.0   
..                           ...  ...             ...   
172                          3.0  ...             NaN   
173                          NaN  ...             NaN   
174                          NaN  ...             4.0   
175                          NaN  ...             NaN   
176                          NaN  ...             0.5   

     The Usual Suspects (1995)  The Mask (1994)  Jaws (1975)  \
0                          NaN              NaN          NaN   
1                          NaN              3.0          NaN   
2                          NaN              NaN          NaN   
3                          3.5              3.5          3.0   
4                          NaN              NaN          NaN   
..                         ...              ...          ...   
172                        NaN              NaN 

In [55]:
siblings_ratings = data_df[data_df['Are you an only child? (1: Yes; 0: No; -1: Did not respond)'] == 0].iloc[:,0:400]
siblings_ratings

The Life of David Gale (2003)  Wing Commander (1999)  \
0                               NaN                    NaN   
1                               NaN                    NaN   
3                               NaN                    NaN   
6                               NaN                    NaN   
7                               3.5                    NaN   
...                             ...                    ...   
1092                            NaN                    NaN   
1093                            3.0                    4.0   
1094                            NaN                    NaN   
1095                            NaN                    NaN   
1096                            NaN                    NaN   

      Django Unchained (2012)  Alien (1979)  \
0                         4.0           NaN   
1                         1.5           NaN   
3                         2.0           NaN   
6                         NaN           NaN   
7                         3.5           2.5   
...                       ...           ...   
1092                      NaN           NaN   
1093                      NaN           NaN   
1094                      NaN           NaN   
1095                      NaN           NaN   
1096                      4.0           NaN   

      Indiana Jones and the Last Crusade (1989)  Snatch (2000)  \
0                                           3.0            NaN   
1                                           NaN            NaN   
3                                           3.0            NaN   
6                                           NaN            NaN   
7                                           3.0            3.0   
...                                         ...            ...   
1092                                        3.5            NaN   
1093                                        4.0            4.0   
1094                                        NaN            NaN   
1095                                        NaN            NaN   
1096                                        2.5            NaN   

      Rambo: First Blood Part II (1985)  Fargo (1996)  \
0                                   NaN           NaN   
1                                   NaN           NaN   
3                                   NaN           NaN   
6                                   NaN           NaN   
7                                   3.0           3.5   
...                                 ...           ...   
1092                                NaN           NaN   
1093                                2.5           NaN   
1094                                NaN           3.5   
1095                                NaN           NaN   
1096                                NaN           3.0   

      Let the Right One In (2008)  Black Swan (2010)  ...  X-Men 2 (2003)  \
0                             NaN                NaN  ...             NaN   
1                             NaN                NaN  ...             NaN   
3                             NaN                4.0  ...             NaN   
6                             NaN                4.0  ...             NaN   
7                             4.0                4.0  ...             2.0   
...                           ...                ...  ...             ...   
1092                          NaN                NaN  ...             NaN   
1093                          3.5                3.5  ...             4.0   
1094                          NaN                NaN  ...             NaN   
1095                          NaN                NaN  ...             NaN   
1096                          NaN                3.5  ...             NaN   

      The Usual Suspects (1995)  The Mask (1994)  Jaws (1975)  \
0                           NaN              NaN          4.0   
1                           NaN              NaN          NaN   
3                           3.0              NaN          NaN   
6                           NaN  

In [56]:
siblings_ratings.reset_index(inplace = True)

In [57]:
siblings_ratings

index  The Life of David Gale (2003)  Wing Commander (1999)  \
0        0                            NaN                    NaN   
1        1                            NaN                    NaN   
2        3                            NaN                    NaN   
3        6                            NaN                    NaN   
4        7                            3.5                    NaN   
..     ...                            ...                    ...   
889   1092                            NaN                    NaN   
890   1093                            3.0                    4.0   
891   1094                            NaN                    NaN   
892   1095                            NaN                    NaN   
893   1096                            NaN                    NaN   

     Django Unchained (2012)  Alien (1979)  \
0                        4.0           NaN   
1                        1.5           NaN   
2                        2.0           NaN   
3                        NaN           NaN   
4                        3.5           2.5   
..                       ...           ...   
889                      NaN           NaN   
890                      NaN           NaN   
891                      NaN           NaN   
892                      NaN           NaN   
893                      4.0           NaN   

     Indiana Jones and the Last Crusade (1989)  Snatch (2000)  \
0                                          3.0            NaN   
1                                          NaN            NaN   
2                                          3.0            NaN   
3                                          NaN            NaN   
4                                          3.0            3.0   
..                                         ...            ...   
889                                        3.5            NaN   
890                                        4.0            4.0   
891                                        NaN            NaN   
892                                        NaN            NaN   
893                                        2.5            NaN   

     Rambo: First Blood Part II (1985)  Fargo (1996)  \
0                                  NaN           NaN   
1                                  NaN           NaN   
2                                  NaN           NaN   
3                                  NaN           NaN   
4                                  3.0           3.5   
..                                 ...           ...   
889                                NaN           NaN   
890                                2.5           NaN   
891                                NaN           3.5   
892                                NaN           NaN   
893                                NaN           3.0   

     Let the Right One In (2008)  ...  X-Men 2 (2003)  \
0                            NaN  ...             NaN   
1                            NaN  ...             NaN   
2                            NaN  ...             NaN   
3                            NaN  ...             NaN   
4                            4.0  ...             2.0   
..                           ...  ...             ...   
889                          NaN  ...             NaN   
890                          3.5  ...             4.0   
891                          NaN  ...             NaN   
892                          NaN  ...             NaN   
893                          NaN  ...             NaN   

     The Usual Suspects (1995)  The Mask (1994)  Jaws (1975)  \
0                          NaN              NaN          4.0   
1                          NaN              NaN          NaN   
2                          3.0              NaN          NaN   
3                          NaN              NaN          NaN   
4                          3.5              4.0          3.5   
..                         ...              ...          ...   
889                        NaN              NaN 

In [59]:
# 0-176 are only child rates, and 177-1070 are the siblings rates
df_append = only_child_ratings._append(siblings_ratings, ignore_index=True)

In [60]:
df_append

index  The Life of David Gale (2003)  Wing Commander (1999)  \
0         2                            NaN                    NaN   
1         4                            NaN                    NaN   
2         5                            3.0                    NaN   
3        22                            2.5                    3.0   
4        33                            NaN                    NaN   
...     ...                            ...                    ...   
1066   1092                            NaN                    NaN   
1067   1093                            3.0                    4.0   
1068   1094                            NaN                    NaN   
1069   1095                            NaN                    NaN   
1070   1096                            NaN                    NaN   

      Django Unchained (2012)  Alien (1979)  \
0                         NaN           NaN   
1                         3.5           NaN   
2                         NaN           2.5   
3                         2.5           3.0   
4                         NaN           NaN   
...                       ...           ...   
1066                      NaN           NaN   
1067                      NaN           NaN   
1068                      NaN           NaN   
1069                      NaN           NaN   
1070                      4.0           NaN   

      Indiana Jones and the Last Crusade (1989)  Snatch (2000)  \
0                                           NaN            NaN   
1                                           0.5            NaN   
2                                           2.0            NaN   
3                                           2.5            1.5   
4                                           NaN            NaN   
...                                         ...            ...   
1066                                        3.5            NaN   
1067                                        4.0            4.0   
1068                                        NaN            NaN   
1069                                        NaN            NaN   
1070                                        2.5            NaN   

      Rambo: First Blood Part II (1985)  Fargo (1996)  \
0                                   NaN           NaN   
1                                   0.5           1.0   
2                                   NaN           NaN   
3                                   3.0           3.0   
4                                   NaN           NaN   
...                                 ...           ...   
1066                                NaN           NaN   
1067                                2.5           NaN   
1068                                NaN           3.5   
1069                                NaN           NaN   
1070                                NaN           3.0   

      Let the Right One In (2008)  ...  X-Men 2 (2003)  \
0                             NaN  ...             NaN   
1                             NaN  ...             2.5   
2                             NaN  ...             4.0   
3                             2.0  ...             4.0   
4                             NaN  ...             4.0   
...                           ...  ...             ...   
1066                          NaN  ...             NaN   
1067                          3.5  ...             4.0   
1068                          NaN  ...             NaN   
1069                          NaN  ...             NaN   
1070                          NaN  ...             NaN   

      The Usual Suspects (1995)  The Mask (1994)  Jaws (1975)  \
0                           NaN              NaN          NaN   
1                           NaN              3.0          NaN   
2                           NaN              NaN          NaN   
3                           3.5              3.5          3.0   
4                           NaN              NaN          NaN   
...                         ...              ..

In [62]:
df_append = df_append.drop('index', axis=1)

In [63]:
df_append

The Life of David Gale (2003)  Wing Commander (1999)  \
0                               NaN                    NaN   
1                               NaN                    NaN   
2                               3.0                    NaN   
3                               2.5                    3.0   
4                               NaN                    NaN   
...                             ...                    ...   
1066                            NaN                    NaN   
1067                            3.0                    4.0   
1068                            NaN                    NaN   
1069                            NaN                    NaN   
1070                            NaN                    NaN   

      Django Unchained (2012)  Alien (1979)  \
0                         NaN           NaN   
1                         3.5           NaN   
2                         NaN           2.5   
3                         2.5           3.0   
4                         NaN           NaN   
...                       ...           ...   
1066                      NaN           NaN   
1067                      NaN           NaN   
1068                      NaN           NaN   
1069                      NaN           NaN   
1070                      4.0           NaN   

      Indiana Jones and the Last Crusade (1989)  Snatch (2000)  \
0                                           NaN            NaN   
1                                           0.5            NaN   
2                                           2.0            NaN   
3                                           2.5            1.5   
4                                           NaN            NaN   
...                                         ...            ...   
1066                                        3.5            NaN   
1067                                        4.0            4.0   
1068                                        NaN            NaN   
1069                                        NaN            NaN   
1070                                        2.5            NaN   

      Rambo: First Blood Part II (1985)  Fargo (1996)  \
0                                   NaN           NaN   
1                                   0.5           1.0   
2                                   NaN           NaN   
3                                   3.0           3.0   
4                                   NaN           NaN   
...                                 ...           ...   
1066                                NaN           NaN   
1067                                2.5           NaN   
1068                                NaN           3.5   
1069                                NaN           NaN   
1070                                NaN           3.0   

      Let the Right One In (2008)  Black Swan (2010)  ...  X-Men 2 (2003)  \
0                             NaN                NaN  ...             NaN   
1                             NaN                0.0  ...             2.5   
2                             NaN                4.0  ...             4.0   
3                             2.0                2.5  ...             4.0   
4                             NaN                3.5  ...             4.0   
...                           ...                ...  ...             ...   
1066                          NaN                NaN  ...             NaN   
1067                          3.5                3.5  ...             4.0   
1068                          NaN                NaN  ...             NaN   
1069                          NaN                NaN  ...             NaN   
1070                          NaN                3.5  ...             NaN   

      The Usual Suspects (1995)  The Mask (1994)  Jaws (1975)  \
0                           NaN              NaN          NaN   
1                           NaN              3.0          NaN   
2                           NaN              NaN          NaN   
3                           3.5  

In [91]:
count = 0
for column in df_append:
     
    # Select column contents by column
    # name using [] operator
    columnSeriesObj = df_append[column]
    only_child = columnSeriesObj[:177].dropna()
    siblings = columnSeriesObj[177:].dropna()
    res = ttest_ind(only_child, siblings)
    if res[1] <= 0.005:
        count += 1

    print('Column Name : ', column)
    print('Column Contents : ', columnSeriesObj.values)
    print('ratings for only child : ', only_child.values)
    print('ratings for siblings : ', siblings.values)
    print('results : ', res)
    print('Count for p value smaller than 0.005 : ', count)

Column Name :  The Life of David Gale (2003)
Column Contents :  [nan nan  3. ... nan nan nan]
ratings for only child :  [3.  2.5 2.5 3.5 1.  0.5 3.  0.5 3.  2.5 3.  2.5 0.  1.5 1.5 2.  3.5]
ratings for siblings :  [3.5 1.5 2.5 1.5 4.  3.5 1.  2.  1.  0.5 3.  1.  2.  0.5 4.  4.  3.  4.
 3.5 3.  3.5 3.  3.5 0.  0.  2.  2.  0.5 0.5 1.  1.5 2.  4.  1.5 3.  0.
 0.5 2.5 3.5 1.5 0.5 2.5 3.  2.  2.5 4.  2.  1.  3.  3.  1.5 3.  1.  2.5
 3. ]
results :  (-0.16658974129116874, 0.8681731382098828, 70.0)
Count for p value smaller than 0.005 :  0
Column Name :  Wing Commander (1999)
Column Contents :  [nan nan nan ... nan nan nan]
ratings for only child :  [3.  3.5 3.  2.  2.  2.5 0.  1.  2.  2.5 2.5 2.  2.  0.  3. ]
ratings for siblings :  [1.5 2.  3.  4.  2.  2.  4.  1.  0.5 0.5 0.  0.  4.  4.  1.5 2.5 3.  3.
 3.5 3.  0.  3.  0.  1.  2.5 1.  0.5 2.  1.  2.  2.5 2.  2.  1.  2.  0.5
 3.5 1.  3.5 2.  2.  2.5 2.  0.5 2.  3.  0.  1.  3.  3.  3.  2.5 2.  4. ]
results :  (0.1690715942806913, 0.8662497368

Column Name :  Shrek (2001)
Column Contents :  [3.5 3.  2.  ... 3.5 2.5 2. ]
ratings for only child :  [3.5 3.  2.  3.  4.  3.  2.5 4.  2.  4.  1.  4.  4.  4.  4.  3.5 3.5 3.5
 3.  4.  3.5 1.5 2.5 4.  3.5 2.  3.  2.  4.  3.5 1.5 4.  3.5 3.  2.5 2.5
 4.  4.  2.5 0.5 1.5 4.  4.  4.  4.  3.  4.  3.5 3.5 3.  1.5 3.  2.5 2.
 2.5 3.  3.  3.  3.  3.5 1.  3.  2.5 3.  3.5 2.5 4.  3.  2.  3.5 3.  3.
 3.  4.  4.  4.  3.5 4.  4.  2.5 1.5 4.  4.  4.  3.5 3.5 3.5 3.5 4.  2.
 4.  3.  1.  3.  2.5 4.  4.  2.  3.5 3.  3.  3.5 4.  2.5 4.  2.5 3.  3.5
 4.  3.5 1.  4.  2.5 3.5 4.  3.5 3.  1.5 1.5 2.5 3.5 3.  2.  3.  4.  3.5
 2.5 3.5 4.  4.  2.  4.  1.  3.  2.5 4.  2.5 3.5 4.  2.5 4.  4.  4.  4.
 4.  2.5 2.5 3.5 2.  4.  3.  4.  4.  2.5]
ratings for siblings :  [3.  3.  0.5 3.5 4.  1.5 4.  4.  2.5 3.  4.  3.5 4.  4.  2.  3.5 4.  4.
 2.5 1.5 4.  4.  2.5 4.  4.  3.  4.  3.  4.  3.  3.  1.5 4.  3.  3.  3.
 1.  3.5 4.  3.  3.5 4.  3.  1.5 2.5 3.  2.5 3.5 2.  2.5 3.  4.  4.  2.5
 4.  4.  1.  1.5 3.5 3.5 3.5 2.5 3

Column Name :  The Evil Dead (1981)
Column Contents :  [nan nan nan ... nan nan nan]
ratings for only child :  [3.  2.5 2.5 2.  4.  2.5 4.  3.5 1.5 3.  3.  3.5 2.  0.  0.  3.5 2.5 1.5
 3.  0.  2.5 2.  1.  3.  2.  2.5 3.5]
ratings for siblings :  [2.  2.5 3.  2.  2.5 2.  3.5 2.5 4.  2.  3.  1.  3.  1.  0.  1.  0.5 1.5
 3.  2.  3.  2.5 4.  0.  2.  1.5 3.5 3.5 3.5 2.5 4.  3.5 3.  1.5 2.5 2.
 3.  1.5 4.  3.5 3.  3.  3.  0.  3.5 3.  2.  3.5 2.5 3.  4.  4.  0.5 4.
 4.  1.  4.  2.  3.  2.  2.5 3.  2.5 4.  3.5 1.5 0.  1.5 3.5 3.  3.  4.
 3.5 2.5 1.5 4.  1.  3.5 3.5 3.5 2.  1.  0.5 3.5 1.  4.  4.  2.5 2.5 3.5
 2.5 2.  0.5 4.  3.  3.  3.5 4.  2.  3.  1.  1.5 3.5 0.  3.5 4.  4.  3.
 3.5]
results :  (-0.9027246687216387, 0.36829189758669156, 134.0)
Count for p value smaller than 0.005 :  3
Column Name :  Gangs of New York (2002)
Column Contents :  [nan nan nan ... nan nan nan]
ratings for only child :  [4.  4.  3.5 3.  4.  2.  3.5 2.5 1.5 4.  3.  2.5 4.  0.  0.  2.  3.5 2.5
 2.5 3.  3.  1.5 3.5 4.

Column Name :  Ghost (1990)
Column Contents :  [nan  2. nan ... nan nan nan]
ratings for only child :  [2.  3.5 3.  2.  2.5 4.  4.  4.  3.  3.  2.5 0.  2.5 2.5 4.  3.5 2.  4.
 2.5 0.5 3.  2.5 1.5 2.5 3.5 0.5 3.5 2.5 3.5 3.  1.  0.5 3.5 3.  0.  3.
 4.  2.5 3.5 4.  2.  2.5 3.  3.5]
ratings for siblings :  [3.5 3.5 2.  2.5 1.5 2.5 3.5 3.  2.  2.  1.5 4.  0.5 4.  4.  3.5 3.5 4.
 0.  3.5 2.  3.  4.  2.5 4.  2.  3.  4.  0.5 1.  2.  3.  4.  3.  2.5 3.5
 3.  3.5 4.  1.  3.5 2.5 3.5 3.  3.5 4.  4.  3.  2.5 3.  3.  1.5 3.  3.5
 3.  3.  3.  4.  3.  1.  3.5 3.  4.  2.  3.  3.5 2.5 2.5 1.5 3.5 3.  1.5
 4.  3.  2.5 3.  2.  2.  3.5 3.  3.5 3.  2.5 3.  3.  4.  3.  3.5 3.5 1.
 1.5 2.5 3.5 3.5 3.  0.  2.  2.  4.  4.  2.5 2.5 3.  3.  2.  2.5 4.  4.
 1.  3.  4.  2.5 2.5 2.5 4.  2.5 2.5 3.  3.  1.  3.5 4.  3.  4.  3.5 2.5
 3.  4.  1.5 3.5 3.  3.  3.  1.  2.  4.  1.5 3.  1.5 0.  3.  3.5 3.  3.
 4. ]
results :  (-0.9395887610499922, 0.34864061113871914, 187.0)
Count for p value smaller than 0.005 :  5
Column

Column Name :  Cinema Paradiso (1988)
Column Contents :  [nan nan nan ... nan nan nan]
ratings for only child :  [3.  4.  3.5 2.5 3.5 3.  2.  2.  2.  0.  1.5 2.  4.  2.  2.5 0.5 2.5 2.
 4. ]
ratings for siblings :  [3.5 1.  3.5 2.  3.  2.5 3.  2.5 3.5 4.  2.5 0.5 4.  3.5 4.  0.5 4.  2.5
 2.5 3.5 3.5 3.5 0.  3.5 4.  1.  4.  4.  4.  0.  3.  3.5 2.  4.  4.  2.5
 1.  3.  2.5 4.  3.5 0.  3.  1.  3.5 3.  4.  4.  2.5 3.  1.5 4.  3.5 4.
 4.  0.  3.  4.  4.  3.5 3.  1.  1.5 3.5 4.  4.  2.5 3.5 4.  3.5 3.5 3.
 1.5 3.5 4.  4.  1.5 2.5 3.5 2.5 2.  3.5]
results :  (-1.525296509268156, 0.13037354490970735, 99.0)
Count for p value smaller than 0.005 :  5
Column Name :  Double Jeopardy (1999)
Column Contents :  [nan  3. nan ... nan nan nan]
ratings for only child :  [3.  3.5 2.  3.5 4.  2.  2.  2.  2.5 0.  2.  3.  2.5 2.  2.5 2.5 3.5 0.5
 2.5 3. ]
ratings for siblings :  [3.5 1.5 2.  3.5 4.  4.  2.5 2.5 2.  4.  2.5 4.  3.5 0.5 3.  2.5 2.  3.5
 2.5 3.5 3.5 3.  3.  3.5 3.  0.5 3.  2.5 0.  3.5 4.  0.5 3.

Column Name :  Inception (2010)
Column Contents :  [4.  3.5 4.  ... 3.  2.5 1. ]
ratings for only child :  [4.  3.5 4.  3.5 4.  3.5 4.  3.  3.  3.5 4.  2.5 3.5 4.  4.  1.  3.5 4.
 3.5 3.5 4.  0.5 4.  4.  2.5 3.5 3.5 4.  3.5 4.  2.5 3.  4.  4.  4.  1.
 3.  2.5 3.  4.  2.  4.  3.5 4.  3.5 4.  2.  3.5 3.  2.5 2.5 4.  4.  2.5
 3.5 4.  4.  3.5 3.  4.  3.5 4.  2.  4.  4.  4.  4.  4.  4.  4.  3.5 2.5
 3.  4.  4.  4.  4.  4.  2.5 4.  2.  4.  3.5 2.5 1.5 3.5 3.5 4.  3.  1.5
 4.  3.  0.  1.  3.  3.  1.  4.  4.  4.  3.5 4.  3.  4.  3.  4.  4.  3.5
 3.  3.5 4.  4.  4.  4.  2.  3.5 4.  3.5 2.5 3.  2. ]
ratings for siblings :  [3.  2.5 3.5 0.5 4.  4.  3.5 3.  3.5 3.5 3.5 4.  4.  1.  3.5 3.  4.  3.5
 4.  3.5 3.5 3.5 4.  4.  3.5 4.  4.  4.  4.  3.5 3.5 4.  3.5 2.  3.5 4.
 2.  2.  3.5 3.5 3.5 3.  4.  3.5 3.5 2.  3.  3.  1.  4.  2.5 4.  3.  1.
 2.5 3.5 3.  4.  3.  4.  3.  2.  3.  2.  4.  4.  4.  4.  3.  3.5 3.5 4.
 3.  3.  4.  4.  4.  4.  3.  2.5 2.  4.  3.  4.  4.  2.  4.  4.  4.  3.
 4.  4.  3.  4.  2

In [81]:
proportion = 10/len(df_append)
proportion

0.009337068160597572

# q7

In [85]:
enjoy_alone = data_df.loc[data_df['Movies are best enjoyed alone (1: Yes; 0: No; -1: Did not respond)'] == 1, 'The Wolf of Wall Street (2013)'].values
enjoy_alone

array([nan, 4. , 3. , 3.5, 3.5, 3. , nan, 4. , 4. , nan, nan, nan, 3. ,
       3.5, 2. , nan, 2.5, 4. , 4. , 4. , 3. , 3.5, 3. , 3. , 4. , 3.5,
       2. , 3. , nan, 3. , nan, 4. , 4. , nan, nan, 3. , nan, 4. , 4. ,
       nan, nan, 3. , 4. , 3.5, 3. , 3.5, 3.5, 3.5, 4. , 3. , 2.5, 3.5,
       nan, 3.5, nan, 3.5, 2. , 3. , 4. , 3.5, nan, nan, 2. , nan, nan,
       3. , 4. , 3.5, nan, 4. , nan, nan, nan, 2.5, 3.5, nan, nan, nan,
       2.5, 3.5, 3.5, nan, 4. , 2. , nan, 3.5, nan, nan, 4. , nan, 2.5,
       3.5, nan, 4. , nan, 2.5, nan, 3.5, 0. , nan, 4. , 3. , 3. , 3.5,
       nan, 3.5, 4. , nan, 4. , nan, 3. , 0.5, 3. , nan, 3.5, nan, 2. ,
       2.5, nan, 3.5, 4. , 3.5, 3.5, 2. , 4. , nan, 1. , 1.5, 4. , nan,
       4. , 1. , nan, nan, nan, 3. , nan, nan, nan, nan, 4. , nan, 4. ,
       2.5, nan, 3.5, nan, 3.5, 3.5, 3.5, 3. , nan, nan, 1.5, 3.5, 3. ,
       nan, nan, 0.5, 4. , 4. , 1.5, nan, 3.5, 3.5, nan, nan, 3.5, nan,
       3. , 3. , nan, 3. , 2.5, 3.5, nan, 3.5, 4. , 2.5, 3. , 4.

In [86]:
enjoy_social = data_df.loc[data_df['Movies are best enjoyed alone (1: Yes; 0: No; -1: Did not respond)'] == 0, 'The Wolf of Wall Street (2013)'].values
enjoy_social

array([nan, nan, 2.5, 1. , 3.5, nan, 4. , nan, 4. , nan, nan, 4. , 2.5,
       1. , nan, nan, 4. , nan, nan, nan, 4. , 3. , nan, 3.5, 3. , nan,
       nan, nan, nan, nan, 3. , 2.5, 2.5, 3. , nan, 4. , nan, nan, nan,
       3.5, 3.5, nan, 3. , nan, nan, 4. , nan, 3. , 3. , nan, nan, 1.5,
       2.5, nan, nan, nan, nan, 4. , 3.5, 4. , 3.5, nan, 3.5, nan, nan,
       3. , nan, nan, nan, 2.5, 3. , nan, 4. , nan, nan, 3.5, nan, 2. ,
       4. , nan, nan, nan, 3.5, 3.5, 3. , 2. , nan, nan, nan, nan, 3.5,
       4. , 1.5, 4. , 4. , nan, 3. , nan, 3. , 1. , nan, 3.5, 2.5, nan,
       2. , nan, 4. , 2. , 3.5, 3. , nan, nan, 3.5, nan, nan, 3.5, nan,
       3.5, nan, nan, 3.5, 3.5, nan, 4. , nan, nan, nan, nan, nan, nan,
       4. , nan, 2.5, nan, 2. , 3.5, 3.5, 2.5, 4. , 2.5, nan, 2.5, 3. ,
       nan, nan, 3.5, 1. , nan, nan, nan, 4. , 4. , 3. , nan, 3. , 3.5,
       2.5, nan, nan, 4. , 4. , nan, 3.5, 2. , 3.5, 4. , nan, nan, 3. ,
       4. , nan, nan, 4. , nan, nan, 3. , nan, 3. , 0.5, nan, 4.

In [87]:
ii = np.isfinite(enjoy_alone)
enjoy_alone = enjoy_alone[ii]
enjoy_alone

array([4. , 3. , 3.5, 3.5, 3. , 4. , 4. , 3. , 3.5, 2. , 2.5, 4. , 4. ,
       4. , 3. , 3.5, 3. , 3. , 4. , 3.5, 2. , 3. , 3. , 4. , 4. , 3. ,
       4. , 4. , 3. , 4. , 3.5, 3. , 3.5, 3.5, 3.5, 4. , 3. , 2.5, 3.5,
       3.5, 3.5, 2. , 3. , 4. , 3.5, 2. , 3. , 4. , 3.5, 4. , 2.5, 3.5,
       2.5, 3.5, 3.5, 4. , 2. , 3.5, 4. , 2.5, 3.5, 4. , 2.5, 3.5, 0. ,
       4. , 3. , 3. , 3.5, 3.5, 4. , 4. , 3. , 0.5, 3. , 3.5, 2. , 2.5,
       3.5, 4. , 3.5, 3.5, 2. , 4. , 1. , 1.5, 4. , 4. , 1. , 3. , 4. ,
       4. , 2.5, 3.5, 3.5, 3.5, 3.5, 3. , 1.5, 3.5, 3. , 0.5, 4. , 4. ,
       1.5, 3.5, 3.5, 3.5, 3. , 3. , 3. , 2.5, 3.5, 3.5, 4. , 2.5, 3. ,
       4. , 3. , 1. , 4. , 3.5, 4. , 3. , 3.5, 3.5, 3.5, 4. , 3. , 4. ,
       2. , 2.5, 3.5, 3.5, 3. , 3. , 3.5, 2.5, 2.5, 4. , 3.5, 2.5, 4. ,
       1. , 4. , 2. , 3. , 4. , 4. , 4. , 4. , 2.5, 4. , 2.5, 2.5, 3. ,
       4. , 3.5, 3.5, 3. , 4. , 4. , 4. , 4. , 4. , 4. , 1.5, 4. , 1.5,
       4. , 4. , 2.5, 4. , 2. , 4. , 2.5, 2. , 0.5, 4. , 3.5, 3.

In [88]:
ii = np.isfinite(enjoy_social)
enjoy_social = enjoy_social[ii]
enjoy_social

array([2.5, 1. , 3.5, 4. , 4. , 4. , 2.5, 1. , 4. , 4. , 3. , 3.5, 3. ,
       3. , 2.5, 2.5, 3. , 4. , 3.5, 3.5, 3. , 4. , 3. , 3. , 1.5, 2.5,
       4. , 3.5, 4. , 3.5, 3.5, 3. , 2.5, 3. , 4. , 3.5, 2. , 4. , 3.5,
       3.5, 3. , 2. , 3.5, 4. , 1.5, 4. , 4. , 3. , 3. , 1. , 3.5, 2.5,
       2. , 4. , 2. , 3.5, 3. , 3.5, 3.5, 3.5, 3.5, 3.5, 4. , 4. , 2.5,
       2. , 3.5, 3.5, 2.5, 4. , 2.5, 2.5, 3. , 3.5, 1. , 4. , 4. , 3. ,
       3. , 3.5, 2.5, 4. , 4. , 3.5, 2. , 3.5, 4. , 3. , 4. , 4. , 3. ,
       3. , 0.5, 4. , 3.5, 3. , 1.5, 3. , 4. , 0. , 3.5, 2.5, 3. , 3. ,
       2.5, 4. , 3.5, 3. , 2. , 3.5, 3. , 4. , 2. , 0.5, 4. , 3.5, 4. ,
       1. , 3.5, 4. , 1. , 3.5, 3. , 3.5, 3. , 4. , 2. , 3. , 4. , 2. ,
       4. , 3.5, 3.5, 3.5, 3.5, 0.5, 4. , 3. , 4. , 3.5, 3.5, 3. , 3.5,
       3.5, 4. , 4. , 4. , 4. , 2.5, 4. , 4. , 3. , 3. , 3.5, 3. , 4. ,
       3. , 0. , 3.5, 3.5, 3.5, 3. , 3. , 2.5, 0.5, 4. , 3.5, 1.5, 4. ,
       3.5, 2.5, 1. , 2.5, 2. , 3. , 2. , 2. , 3.5, 3.5, 1. , 4.

In [89]:
pg.ttest(enjoy_alone, enjoy_social, correction=False) #Welch’s t-test as the sample size is not equal

T  dof alternative     p-val          CI95%   cohen-d   BF10  \
T-test  1.567874  661   two-sided  0.117389  [-0.03, 0.25]  0.123934  0.293   

           power  
T-test  0.346866

# Two-Sample T-Test with Statsmodels
The p-value of the test comes out to be equal to 0.117, which is greater than the significance level alpha (that is, 0.005). This implies that we can say that the people who like to watch movies socially enjoy ‘The Wolf of Wall Street (2013)’ is statistically not different from the the people who prefer to watch them alone. 



# q10

In [93]:
Star_Wars_cols = [col for col in data_df.columns if 'Star Wars' in col]
print(Star_Wars_cols)

['Star Wars: Episode IV - A New Hope (1977)', 'Star Wars: Episode II - Attack of the Clones (2002)', 'Star Wars: Episode V - The Empire Strikes Back (1980)', 'Star Wars: Episode 1 - The Phantom Menace (1999)', 'Star Wars: Episode VII - The Force Awakens (2015)', 'Star Wars: Episode VI - The Return of the Jedi (1983)']


In [94]:
M1 = pd.to_numeric(data_df['Star Wars: Episode IV - A New Hope (1977)'],errors='coerce').values
M2 = pd.to_numeric(data_df['Star Wars: Episode II - Attack of the Clones (2002)'],errors='coerce').values
M3 = pd.to_numeric(data_df['Star Wars: Episode V - The Empire Strikes Back (1980)'],errors='coerce').values
M4 = pd.to_numeric(data_df['Star Wars: Episode 1 - The Phantom Menace (1999)'],errors='coerce').values
M5 = pd.to_numeric(data_df['Star Wars: Episode VII - The Force Awakens (2015)'],errors='coerce').values
M6 = pd.to_numeric(data_df['Star Wars: Episode VI - The Return of the Jedi (1983)'],errors='coerce').values

In [95]:
M1 = M1[np.isfinite(M1)] # only keep the finite elements (not infinity or NaN)
M2 = M2[np.isfinite(M2)] # only keep the finite elements (not infinity or NaN)
M3 = M3[np.isfinite(M3)] # only keep the finite elements (not infinity or NaN)
M4 = M4[np.isfinite(M4)] # only keep the finite elements (not infinity or NaN)
M5 = M5[np.isfinite(M5)] # only keep the finite elements (not infinity or NaN)
M6 = M6[np.isfinite(M6)] # only keep the finite elements (not infinity or NaN)

In [99]:
#%% 3 Reformatting - V1: Format data into a representation to work with
combinedData = np.transpose(np.array([M1,M2,M3,M4,M5,M6], dtype=object)) # array of arrays
    
#%% 4a) Extrastriate cortex: Doing the actual specialized analyses
# Descriptive statistics - we are looking for very special numbers that
# capture the essence of the entire dataset - the typical number (central
# tendency) and the dispersion. Typically mean and SD. 
numMovies = 6
# Initialize container to store descriptives:
descriptivesContainer = np.empty([numMovies,4])
descriptivesContainer[:] = np.NaN 

# 1. Element-wise:
for ii in range(numMovies):
    descriptivesContainer[ii,0] = np.mean(combinedData[ii]) # mu
    descriptivesContainer[ii,1] = np.std(combinedData[ii]) # sigma
    descriptivesContainer[ii,2] = len(combinedData[ii]) # n
    descriptivesContainer[ii,3] = descriptivesContainer[ii,1]/np.sqrt(descriptivesContainer[ii,2]) # sem

In [100]:
descriptivesContainer

array([[2.97069597e+00, 1.02266542e+00, 5.46000000e+02, 4.37660399e-02],
       [2.44746717e+00, 1.05392065e+00, 5.33000000e+02, 4.56503710e-02],
       [3.09611452e+00, 9.37715339e-01, 4.89000000e+02, 4.24049534e-02],
       [2.48785872e+00, 1.10255896e+00, 4.53000000e+02, 5.18027387e-02],
       [3.11752577e+00, 9.46835240e-01, 4.85000000e+02, 4.29935735e-02],
       [3.02980132e+00, 9.75242860e-01, 4.53000000e+02, 4.58209067e-02]])